### DISCUSSION TITLE
"Additional Latency" to make hot backup

### DISCUSSION THREAD

**Discussion by Chris5342** [Ysurac/openmptcprouter#2357] [Q&A]
Am I right in thinking that "Additional Latency" (Network -> Interfaces -> WAN? -> Edit)  is the only way to tell OMR to de-prefer a WAN interface (hot backup) ?

Also is  "Additional Latency" only applied after a reboot ?

What am I trying to do ?

I'm trying to stop OMR putting my slow B/W low latency ADSL into bat and prefer the Starlink interface unless Starlink is have problems.

Why does this matter ?

During a video call OMR seems to keep putting the ADSL in and out (as the ADSL and Starlink latency are similar I assume).

Snag is when Starlink is put back there is a dip and seemingly the video call goes weird. See image:
![Screenshot 2022-05-19 at 09 17 38](https://user-images.githubusercontent.com/105393840/169686508-e8f6f750-cfc6-4bcb-9fa8-9f832011ff27.jpg)
Notice that the upload link via Starlink seems stable but the download keeps switching between Starlink and ADSL but that there are gaps before Starlink takes over.

What have I tried ?

Making ADSL backup - no luck that kills everything

Setting ADSL "Additional Latency" to 300 seems to work and that stops ADSL being used

Setting ADSL "Additional Latency" to 80 doesn't seem to help

Is there a way for me to see what OMR is doing (logs etc) ?
Eg why is it putting ADSL into  bat ?

**Comment by Ysurac** [comment]
Additional Latency is only here for tests, should not be used.
Connections are used based on latency, when connection is fully used this used next connection with higher latency. Shouldn't need any customization in most case.

**Reply by Chris5342** [reply]
@Ysurac   Thanks for quick reply.

> Additional Latency is only here for tests, should not be used.

I'm getting desperate !

> Connections are used based on latency, when connection is fully used this used next connection with higher latency. 

But that means it could see saw between ADSL and Starlink as they have similar latency.
In fact I've seen latency go up with traffic so we could well expect it to oscillate back and forth.

I realise it should not matter which interface carries the traffic but as the graph above shows the traffic doesn't seem to switch back to Starlink seamlessly (note the gaps).

It's not an academic point as it does seem to effect Video Calls.

Also I'm not the only one: https://github.com/Ysurac/openmptcprouter/discussions/2249

**Reply by Ysurac** [reply]
I don't understand what is the problem.
Try with "omr-test-speed" via SSH on the router, you should see that all connections are used when needed.

**Reply by Chris5342** [reply]
@Ysurac I continue to monitor. Occasionally I get odd things eg:

`Thu May 26 12:35:19 2022 user.notice post-tracking-post-tracking: wan1 (wan1) switched off because No access to server API`
`Thu May 26 12:35:19 2022 user.notice post-tracking-post-tracking: Delete default route via 192.168.1.254 dev wan1`

`Thu May 26 12:36:00 2022 user.notice post-tracking-post-tracking: wan1 (wan1) switched up`

wan1 is supposedly slow stable ADSL.
What does "switched off because No access to server API" mean ?

Also I captured this:
![Screenshot 2022-05-26 at 13 11 40](https://user-images.githubusercontent.com/105393840/170489712-5d613936-e929-4058-aaff-fb2cb38a4877.jpg)

Traffic is simulated video conference, approx 1.8Mbs down 1.4Mbps up.

It seems to show that ADSL was not put into bat when Starlink had a 4s outage - but ADSL may have been coincidentally down at the time:

`Thu May 26 12:09:49 2022 user.notice post-tracking-post-tracking: wan1 (wan1) switched off because No access to server API`
`Thu May 26 12:09:49 2022 user.notice post-tracking-post-tracking: Delete default route via 192.168.1.254 dev wan1`

`Thu May 26 12:09:59 2022 user.notice post-tracking-post-tracking: wan1 (wan1) switched up.`

Could I add my own more detailed logging ?
If so where is the code that decide what interface to use ?

**Reply by Ysurac** [reply]
The no access to server API, mean it don't have answer from VPS API (there is an API on server side).
You can try to disable HTTP API test in System->OpenMPTCProuter, "advanced settings" tab and disable http ping to API checkbox
But API should always answer...

**Reply by Chris5342** [reply]
@Ysurac 
Could this be what's happening ?

Starlink goes down.
OMR sends all the traffic to ASDL
ADSL saturates
OMR ADSL http API test fails (because the link is saturated)
ADSL is dropped

(A cascade falilure)

I will continue to monitor

**Reply by Ysurac** [reply]
Possible, you can try to enable SQM and set download and upload speed on ADSL link via the wizard. This should help to doesn't have ADSL link saturated.
